webplugin/js/app/zendesk-chat-service.js

```diff
@@ -6,9 +6,9 @@ function ZendeskChatService() {
     var ZENDESK_SDK_INITIALIZED = false;
     var ZENDESK_CHAT_SDK_KEY = "";
 
-    _this.init = function (zendeskChatSdkKey) {
+    _this.init = function (zendeskChatSdkKey, preChatData) {
         ZENDESK_CHAT_SDK_KEY = zendeskChatSdkKey;
-        _this.loadZopimSDK();
+        _this.loadZopimSDK(preChatData);
```

Comment:
see if we can store these user details on class level and then use it from there.
no need to pass on each level to all the middle functions